File: EngSoft.Jogo.Patos/src/engsoft/jogo/patos/Nao_Voa.java
package engsoft.jogo.patos;

// Classe concreta
// Polimorfismo
public class Nao_Voa implements Padrao_Voaveis {


    public Nao_Voa() { }

    public String voar() {
        return "Esse pato não Voa. Velocidade: " + getVelocidade();
    }

    public double getVelocidade() {
        return 0;
    }

}

File: EngSoftTypeScript/src/NaoVoa.ts
import {PadraoVoaveis} from "./PadraoVoaveis";

export class NaoVoa implements PadraoVoaveis {

        public voar(): string {
            return "Esse pato não Voa. Velocidade: " + this.getVelocidade();
        }

        public getVelocidade(): number {
            return 0;
        }

}
